libraries/AP_HAL_ChibiOS/hwdef/ZeroOneX6/hwdef.dat

```diff
@@ -0,0 +1,335 @@
+
+# SPI2 -IMU1 ICM45686
+PI1 SPI2_SCK SPI2
+PI2 SPI2_MISO SPI2
+PI3 SPI2_MOSI SPI2
+PH5 SP2_CS1 CS
+PA10 SP2_DRDY2 INPUT
+
+# SPI3 -IMU2 BMI088
+PB2  SPI3_MOSI SPI3
+PC10 SPI3_SCK  SPI3
+PC11 SPI3_MISO SPI3
+PI4 SP3_CS1 CS
+PI8 SP3_CS2 CS
+PI6 SP3_DRDY1 INPUT
+PI7 SP3_DRDY2 INPUT GPIO(93)
+define SP3_DRDY2 93
+
+# SPI4 - unused
+#PE12 SPI4_SCK SPI4
+#PE13 SPI4_MISO SPI4
+#PE14 SPI4_MOSI SPI4
+#PF3  SP4_DRDY1 INPUT
+PH15 SP4_CS1 CS
+
+# SPI5 - FRAM
+PF7  SPI5_SCK  SPI5
+PH7  SPI5_MISO SPI5
+PF11 SPI5_MOSI SPI5
+PG7  FRAM_CS CS
+
+# SPI6 - external1 (disabled to save DMA channels)
+# PB3  SPI6_SCK SPI6
+# PA6  SPI6_MISO SPI6
+# PG14 SPI6_MOSI SPI6
+# PI10   EXT1_CS CS
+
+# PWM output pins
+PI0  TIM5_CH4 TIM5 PWM(1) GPIO(50) BIDIR
+PH12 TIM5_CH3 TIM5 PWM(2) GPIO(51)
+PH11 TIM5_CH2 TIM5 PWM(3) GPIO(52) BIDIR
+PH10 TIM5_CH1 TIM5 PWM(4) GPIO(53)
+PD13 TIM4_CH2 TIM4 PWM(5) GPIO(54) BIDIR
+PD14 TIM4_CH3 TIM4 PWM(6) GPIO(55) BIDIR
+
+# Pin for PWM Voltage Selection, 0 means 3.3v, 1 means 5v
+PG6 PWM_VOLT_SEL OUTPUT HIGH GPIO(3)
+define HAL_GPIO_PWM_VOLT_PIN 3
+define HAL_GPIO_PWM_VOLT_3v3 0
+
+# we need to disable DMA on the last 2 FMU channels
+# as timer 12 doesn't have a TIMn_UP DMA option
+PH6  TIM12_CH1 TIM12 PWM(7) GPIO(56) NODMA
+PH9  TIM12_CH2 TIM12 PWM(8) GPIO(57) NODMA
+
+# GPIOs
+PE11 FMU_CAP1 INPUT GPIO(58)
+PC0 NFC_GPIO INPUT GPIO(60)
+
+# CAN bus
+PD0  CAN1_RX CAN1
+PD1  CAN1_TX CAN1
+
+PB12 CAN2_RX CAN2
+PB13 CAN2_TX CAN2
+
+# I2C buses
+
+# I2C1, GPS+MAG
+PB9 I2C1_SDA I2C1
+PB8 I2C1_SCL I2C1
+
+# I2C2, GPS2+MAG
+PF1 I2C2_SCL I2C2
+PF0 I2C2_SDA I2C2
+PG5 DRDY1_BMP388 INPUT
+
+# I2C3, MS5611, external
+PA8 I2C3_SCL I2C3
+PH8 I2C3_SDA I2C3
+
+# I2C4 internal
+PF14 I2C4_SCL I2C4
+PF15 I2C4_SDA I2C4
+
+# order of I2C buses
+I2C_ORDER I2C4 I2C1 I2C2 I2C3
+define HAL_I2C_INTERNAL_MASK 1
+
+# heater
+PB10 HEATER_EN OUTPUT LOW GPIO(80)
+define HAL_HEATER_GPIO_PIN 80
+
+# Setup the IMU heater
+define HAL_HAVE_IMU_HEATER 1
+define HAL_IMU_TEMP_DEFAULT 45
+define HAL_IMUHEAT_P_DEFAULT 50
+define HAL_IMUHEAT_I_DEFAULT 0.07
+
+# armed indication
+PE6 nARMED OUTPUT HIGH
+
+# power enable pins
+PC13 VDD_3V3_SD_CARD_EN OUTPUT HIGH
+PI11 VDD_3V3_SENSORS1_EN OUTPUT HIGH
+PF4  VDD_3V3_SENSORS2_EN OUTPUT HIGH
+PE7  VDD_3V3_SENSORS3_EN OUTPUT HIGH
+PG8  VDD_3V3_SENSORS4_EN OUTPUT HIGH
+
+# start peripheral power on
+PG10 nVDD_5V_HIPOWER_EN OUTPUT LOW
+PG4  nVDD_5V_PERIPH_EN  OUTPUT LOW
+
+# Control of Spektrum power pin
+PH2 SPEKTRUM_PWR OUTPUT HIGH GPIO(73)
+define HAL_GPIO_SPEKTRUM_PWR 73
+
+# Spektrum Power is Active High
+define HAL_SPEKTRUM_PWR_ENABLED 1
+
+# power sensing
+PE15 VDD_5V_PERIPH_nOC INPUT PULLUP
+PF13 VDD_5V_HIPOWER_nOC INPUT PULLUP
+
+PG1 VDD_BRICK_nVALID INPUT PULLUP
+PG2 VDD_BRICK2_nVALID INPUT PULLUP
+PG3 VDD_BRICK3_nVALID INPUT PULLUP
+
+# microSD support
+PD6  SDMMC2_CK SDMMC2
+PD7  SDMMC2_CMD SDMMC2
+PB14 SDMMC2_D0 SDMMC2
+PB15 SDMMC2_D1 SDMMC2
+PG11 SDMMC2_D2 SDMMC2
+PB4  SDMMC2_D3 SDMMC2
+define FATFS_HAL_DEVICE SDCD2

```

Comment:
I don't think this is required in the main firmware

Reply:
Hello, it needs to be retained, otherwise, there will be some compilation errors...